tools/nl/detection_evals/golden_curation/generate_vars.py

```diff
@@ -0,0 +1,370 @@
+        top_level_topics_to_remove.add(curr_node)
+      curr_node_str = ''
+      for _ in range(curr_lvl):
+        curr_node_str += '-->'
+      curr_node_str += curr_node
+      topic_strs[topic].append(curr_node_str)
+      # if the node is not found in the topic cache, assume it is a sv
+      if not topic_cache.get(curr_node):
+        topic_svs.add(curr_node)
+      next_nodes.extend([
+          (curr_lvl + 1, n) for n in topic_cache.get(curr_node, [])
+      ])
+  # remove top level topics that are found within other topics
+  filtered_topic_strs = {
+      k: v for k, v in topic_strs.items() if k not in top_level_topics_to_remove
+  }
+  return filtered_topic_strs, topic_svs
+
+
+def get_vars_from_nl_response(query_response) -> List[VarInfo]:
+  """
+  Extracts variable information from a natural language query response.
+  This function processes the debug information in the provided query response to extract variable dcids and their associated cosine similarity scores.
+
+  Args:
+    query_response (dict): The response object from a natural language query, expected to contain debug information with variable details.
+
+  Returns:
+    List[VarInfo]: A list of VarInfo objects, each containing a dcid and its corresponding cosine similarity score.
+  """
+  debug_info = query_response.get('debug', {})
+  var_set = set()
+  var_to_score = {}
+  debug_sv_info = debug_info.get('sv_matching', {})
+  for idx, v in enumerate(debug_sv_info.get('SV', [])):
+    var_to_score[v] = debug_sv_info['CosineScore'][idx]
+  for v in debug_info.get('counters', {}).get('INFO',
+                                              {}).get('filtered_svs', [[]])[0]:
+    var_set.add(v)
+  for multi in debug_sv_info.get('MultiSV', {}).get('Candidates', []):
+    for p in multi.get('Parts', []):
+      scores = p.get('CosineScore', [])
+      for idx, v in enumerate(p.get('SV', [])):
+        if scores[idx] < 0.7:
+          break
+        var_set.add(v)
+        var_to_score[v] = var_to_score.get(v, scores[idx])
+  var_result = []
+  for v in var_set:
+    var_result.append(VarInfo(v, var_to_score[v]))
+  return var_result
+
+
+def get_var_dcids(var_string, svg_list):
+  """
+  Retrieve and trim variable DCIDs associated with a given variable string and a list of SVGs.
+
+  Args:
+    var_string (str): The variable string to query for associated variables.
+    svg_list (list of str): List of SVGs to search within for associated variables.
+
+  Returns:
+    list: A list of variable DCIDs associated with the input variable string and SVGs.
+  """
+  var_list = []
+  # query the index for each svg to find all the associated vars for the var string
+  for svg in svg_list:
+    embedding_idx = _SVG_TO_EMBEDDING_IDX.get(svg)
+    idx_param = f'&idx={embedding_idx}' if embedding_idx else ''
+    request_query = f'{var_string} {_DEFAULT_PLACE}'
+    nl_response = requests.post(
+        f'http://localhost:8080/api/explore/detect?q={request_query}{idx_param}',
```

Comment:
Is there a benefit to using `/api/explore/detect` over `/api/nl/search-vector`?

Reply:
I don't think so, I just chose detect because I didn't know the search-vector endpoint existed before ..